### PR TITLE
Feature authorization

### PR DESCRIPTION
**Pull request by Tchilo**
- Install CanCanCan in your project.
- Add the role column to the user's table.
- Add delete functionality to post.
-  Add the "Delete" button to the view.
- Add comment deletion functionality.
- Add  CanCanCan for authorization.
-  Add the "Delete" button to the view and make it visible only by authorized users.

### CODE REVIEW

File: app/controllers/likes_controller.rb [hunk 1]
```diff
@@ -7,6 +9,22 @@ def create
...
+  #   like.save!
+
+  #   if like.save
+  #     flash[:success] = 'Post Liked!'
+  #     redirect_to user_post_path
+  #   else
+  #     flash.now[:error] = 'Like not added...'
+  #     redirect_to post_path(@post)
+  #   end
+  # end
```
Comment (Dipeshtwis) [inline app/controllers/likes_controller.rb]: - [x] I would recommend removing the commented code to make the codebase cleaner.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Tchilo & @KingsleyIbe, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
:+1: THe Application is working as expected. :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

- [ ] In the README file, also mention the steps like `rails db:setup` to setup and migrate the database.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/likes_controller.rb (hunk 1 above)
Comment (Tchilo) [reply app/controllers/likes_controller.rb]: changes made 😃

**Review by bimbolabuari — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**